go/registry/api/api.go

```diff
@@ -995,66 +984,53 @@ func exactlyOneTrue(conds ...bool) bool {
 	return total == 1
 }
 
-// VerifyRegisterRuntimeArgs verifies arguments for RegisterRuntime.
-func VerifyRegisterRuntimeArgs( // nolint: gocyclo
+// VerifyRuntime verifies the given runtime.
+func VerifyRuntime( // nolint: gocyclo
 	params *ConsensusParameters,
 	logger *logging.Logger,
-	sigRt *SignedRuntime,
+	rt *Runtime,
 	isGenesis bool,
 	isSanityCheck bool,
-) (*Runtime, error) {
-	var rt Runtime
-	if sigRt == nil {
-		return nil, ErrInvalidArgument
-	}
-
-	var ctx signature.Context
-	switch isGenesis {
-	case true:
-		ctx = RegisterGenesisRuntimeSignatureContext
-	case false:
-		ctx = RegisterRuntimeSignatureContext
+) error {
+	if rt == nil {
+		return fmt.Errorf("%w: no runtime given", ErrInvalidArgument)
 	}
 
-	if err := sigRt.Open(ctx, &rt); err != nil {
-		logger.Error("RegisterRuntime: invalid signature",
-			"signed_runtime", sigRt,
-		)
-		return nil, ErrInvalidSignature
-	}
-	if err := sigRt.Signed.Signature.SanityCheck(rt.EntityID); err != nil {
-		logger.Error("RegisterRuntime: invalid argument(s)",
-			"signed_runtime", sigRt,
-			"runtime", rt,
-			"err", err,
-		)
-		return nil, ErrInvalidArgument
-	}
 	if err := rt.ValidateBasic(!isGenesis && !isSanityCheck); err != nil {
 		logger.Error("RegisterRuntime: invalid runtime descriptor",
 			"runtime", rt,
 			"err", err,
 		)
-		return nil, fmt.Errorf("%w: %s", ErrInvalidArgument, err)
+		return fmt.Errorf("%w: %s", ErrInvalidArgument, err)
 	}
 
 	if rt.ID.IsTest() && !params.DebugAllowTestRuntimes {
 		logger.Error("RegisterRuntime: test runtime registration not allowed",
 			"id", rt.ID,
 		)
-		return nil, fmt.Errorf("%w: test runtime not allowed", ErrInvalidArgument)
+		return fmt.Errorf("%w: test runtime not allowed", ErrInvalidArgument)
 	}
 
 	if err := rt.Genesis.SanityCheck(isGenesis); err != nil {
-		return nil, err
+		return err
+	}
+
+	// Make sure the specified runtime governance model is allowed.
+	if params.EnableRuntimeGovernanceModels == nil {
+		// No runtime governance models are allowed.
+		return fmt.Errorf("%w: no runtime governance models are enabled", ErrForbidden)
+	}
+	if allowed, exists := params.EnableRuntimeGovernanceModels[rt.GovernanceModel]; !exists || !allowed {
```

Comment:
The default value in case a key doesn't exist is `false` so just do:
```suggestion
	if !params.EnableRuntimeGovernanceModels[rt.GovernanceModel] {
```